cmake/app/boilerplate.cmake

```diff
@@ -210,6 +270,39 @@ foreach(root ${BOARD_ROOT})
   if(BOARD_DIR AND NOT (${root} STREQUAL ${ZEPHYR_BASE}))
     set(USING_OUT_OF_TREE_BOARD 1)
   endif()
+
+  set(shield_dir ${root}/boards/shields)
+  # Match the .overlay files in the shield directories to make sure we are
+  # finding shields, e.g. x_nucleo_iks01a1/x_nucleo_iks01a1.overlay
+  file(GLOB_RECURSE shields_refs_list
+    RELATIVE ${shield_dir}
+    ${shield_dir}/*/*.overlay
+    )
+
+  # The above gives a list like
+  # x_nucleo_iks01a1/x_nucleo_iks01a1.overlay;x_nucleo_iks01a2/x_nucleo_iks01a2.overlay
+  # we construct a list of shield names by extracting file name and
+  # removing the extension.
+  foreach(shield_path ${shields_refs_list})
+    get_filename_component(shield ${shield_path} NAME_WE)
+
+    list(APPEND SHIELD_LIST ${shield})
+    if(DEFINED SHIELD)
+    if(${shield} STREQUAL ${SHIELD})
```

Comment:
With Kconfig, it was possible to enable multiple shields through Kconfig.

To retain this functionality we need to check if ${shield} is in the list ${SHIELD}.

... and possibly rename from SHIELD to SHIELDS.

Reply:
How would we specify multiple shields, as a 'space separated' list?

`-DSHIELD="frdm_kw41z x_nucleo_iks01a1"`

Reply:
Yes, CONF_FILE should be able to demonstrate how to do it if you see any problems.

Reply:
> Yes, CONF_FILE should be able to demonstrate how to do it if you see any problems.

Ok, should now handle this case.